### PR TITLE
build: print instance being used

### PR DESCRIPTION
**Pull request by crazy-max**
~needs https://github.com/moby/buildkit/pull/3378~

Display current builder being used when building. Can help users who need to know the type of builder in use during their build but also for log reports:

```
$ docker buildx build .
[+] Building 4.8s (11/16)                                                               docker-container:builder2
 => [internal] load build definition from Dockerfile                                                         0.1s
 => => transferring dockerfile: 2.93kB                                                                       0.1s 
 => [internal] load .dockerignore                                                                            0.1s 
 => => transferring context: 45B                                                                             0.1s 
 => resolve image config for docker.io/docker/dockerfile-upstream:1.5.0                                      0.6s
 => CACHED docker-image://docker.io/docker/dockerfile-upstream:1.5.0@sha256:bda6ac9f61f2b676331acfd656a07bc  0.0s
 => => resolve docker.io/docker/dockerfile-upstream:1.5.0@sha256:bda6ac9f61f2b676331acfd656a07bcd55b369143a  0.0s 
 => [internal] load metadata for docker.io/tonistiigi/xx:1.1.2                                               2.4s 
 => [internal] load metadata for docker.io/library/golang:1.19-alpine                                        2.4s 
 => [auth] library/golang:pull token for registry-1.docker.io                                                0.0s
 => [auth] tonistiigi/xx:pull token for registry-1.docker.io                                                 0.0s
 => [xx 1/1] FROM docker.io/tonistiigi/xx:1.1.2@sha256:9dde7edeb9e4a957ce78be9f8c0fbabe0129bf5126933cd35748  0.0s
 => => resolve docker.io/tonistiigi/xx:1.1.2@sha256:9dde7edeb9e4a957ce78be9f8c0fbabe0129bf5126933cd3574888f  0.0s 
 => CANCELED [internal] load build context                                                                   1.5s 
 => => transferring context: 20.57kB                                                                         1.5s 
 => [golatest 1/1] FROM docker.io/library/golang:1.19-alpine@sha256:2381c1e5f8350a901597d633b2e517775eeac7a  0.1s 
 => => resolve docker.io/library/golang:1.19-alpine@sha256:2381c1e5f8350a901597d633b2e517775eeac7a6682be392  0.0s 
ERROR: failed to solve: Canceled: context canceled
```

```
$ BUILDKIT_PROGRESS=plain docker buildx build .
#0 building with "builder2" instance using docker-container driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 2.93kB 0.0s done
#1 DONE 0.1s

#2 [internal] load .dockerignore
#2 transferring context: 45B 0.0s done
#2 DONE 0.1s

#3 resolve image config for docker.io/docker/dockerfile-upstream:1.5.0
#3 DONE 0.7s

#4 docker-image://docker.io/docker/dockerfile-upstream:1.5.0@sha256:bda6ac9f61f2b676331acfd656a07bcd55b369143ab7db66bdf93b619da3e183
#4 resolve docker.io/docker/dockerfile-upstream:1.5.0@sha256:bda6ac9f61f2b676331acfd656a07bcd55b369143ab7db66bdf93b619da3e183
#4 resolve docker.io/docker/dockerfile-upstream:1.5.0@sha256:bda6ac9f61f2b676331acfd656a07bcd55b369143ab7db66bdf93b619da3e183 0.0s done
#4 CACHED

#5 [internal] load metadata for docker.io/library/golang:1.19-alpine
^C#5 CANCELED

#6 [internal] load metadata for docker.io/tonistiigi/xx:1.1.2
#6 CANCELED
ERROR: failed to solve: Canceled: context canceled
```

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: commands/util.go [hunk 1]
```diff
@@ -475,3 +476,31 @@ func boot(ctx context.Context, ngi *nginfo) (bool, error) {
...
+		if err != nil {
+			return err
+		}
+	}
+	if ng.Driver == "" {
+		ng.Driver = "docker"
+	}
+	return progress.Wrap(fmt.Sprintf("[internal] building with %q instance using %s driver", ng.Name, ng.Driver), l, func(sub progress.SubLogger) error {
+		// noop
+		return nil
```
Comment (jedevc) [inline commands/util.go:504]: I think we should be able to do this with a `oneOffProgress` function: https://github.com/moby/buildkit/blob/master/exporter/containerimage/writer.go#L561-L576 to be more consistent with how we already do these kind of one-off messages.

Although, I'm not sure why we have so many identical private `oneOffProgress` functions, we should definitely be able to refactor all of them together.

**Comment by crazy-max** [comment]
> Although, I'm not sure why we have so many identical private `oneOffProgress` functions, we should definitely be able to refactor all of them together.

Opened https://github.com/moby/buildkit/pull/3006 but let's do that in a follow-up when it's merged. LGTY otherwise?

PTAL @tonistiigi

**Review by jedevc — APPROVED** [review]
LGTM :tada:

I'd prefer to replace the `progress.Wrap` call with a `oneOffProgress` function just so we don't forget to `grep` it out when the buildkit PR merges, but not a blocker :+1:

**Comment by crazy-max** [comment]
needs https://github.com/moby/buildkit/pull/3378

File: go.mod [hunk 2]
```diff
@@ -138,39 +140,41 @@ require (
...
-	sigs.k8s.io/structured-merge-diff/v4 v4.1.2 // indirect
-	sigs.k8s.io/yaml v1.2.0 // indirect
+	k8s.io/klog/v2 v2.80.1 // indirect
+	k8s.io/utils v0.0.0-20221108210102-8e77b1f39fe2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 replace (
+	github.com/moby/buildkit => github.com/crazy-max/buildkit v0.7.1-0.20230210175928-b99541fb07b6 // https://github.com/crazy-max/buildkit/tree/progress-desc
```
Comment (crazy-max) [inline go.mod:177]: needs https://github.com/moby/buildkit/pull/3378

**Review by jedevc — APPROVED** [review]
LGTM - just a note, this doesn't work with the remote controller, which has it's own progress setup.

However, that's fine if we can open a tracking issue after this merges, I think some more general refactoring of how we do progress is necessary to get a patch that makes more sense for this.

**Pull request merged**